src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -78,6 +81,18 @@ object Main {
 
     val Parsed(settings, residual, errors) = Settings.parse(args)
 
+    mainImpl(settings.withAbsolutePaths(Paths.get(".").toAbsolutePath.toFile), errors, startTime)
+  }
+
+  def nailMain(context: NGContext): Unit = {
+    val startTime = System.currentTimeMillis
+
+    val Parsed(settings, residual, errors) = Settings.parse(context.getArgs)
+
+    mainImpl(settings.withAbsolutePaths(new File(context.getWorking`Directory)), errors, startTime)
```

Comment:
What strange magic is this?:
```scala
context.getWorking`Directory
```
I have to assume the '`' in getWorkingDirectory is a typo.

Reply:
Oops! Must have happened errantly during a save after testing!